### ISSUE TITLE
`item_suffix` and `item_prefix` are not displayed in some menus

### ISSUE THREAD

**Issue by Uwuewsky** [CleverRaven/Cataclysm-DDA#72193]
### Describe the bug

Prefixes and suffixes presented in #71702 are not displayed in the crafting menu and in dialogue options.

### Attach save file

N/A; requires game modification

### Steps to reproduce

1. Add to the end of the `data/json/flags.json` file:
```json
  {
    "id": "PREFIX_DEBUG",
    "type": "json_flag",
    "item_prefix": "[CUSTOM PREFIX] "
  }
```
2. Extend some item (`data/json/items/comestibles/dairy.json`) with this flag:
```json
  {
    "type": "COMESTIBLE",
    "id": "milk",
    "name": { "str_sp": "milk" },
    ...
    "flags": [ "EATEN_COLD", "PREFIX_DEBUG" ],
    ...
  }
```
3. Spawn this item and see that the prefix is actually added:
![i](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/af600970-2205-42c8-804b-92890b541328)
4. Unlock all recepies through debug menu.
5. [FIXED #72202]
Open crafting menu. Note that the item in the left column (and in the right column when viewing nested recipes) is displayed without its prefix. Same for items in the middle column - tools and components - if you add a flag to the microwave:
![y](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/caa3a6a8-7b00-4ede-b990-f3dc96309c49)
6. Teleport to the refugee center and offer food to the beggar. In the dialog it is also displayed without a prefix.
![q](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/a0628a61-a49e-44f3-867d-baba75a24d42)

### Expected behavior

I expect that the item in the left and middle column will be displayed with prefixes/suffixes. Like this:
![y1](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/ecc74259-abb4-49ee-9ea0-12caa63946db)

### Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Debian; Description: Debian GNU/Linux 12 (bookworm); Release: 12; Codename: bookworm; 
- Game Version: 9d6677c [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context
<details><summary>Fixed crafting menu</summary>
<p>

![a](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/555bd6b8-b07f-43d9-b036-f75cdf9d8ddf)
![b](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/79a3e1d0-e174-4808-9e79-aea84379de25)

</p>
</details> 

Building menu is ok
![e](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/cbdf8786-c780-4c97-873d-09a0aa568068)

**Comment by Uwuewsky** [comment]
Prefixes are not displayed in the `e` menu lines:
"Can be stored in"
![desc](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/263f59d7-8758-4afa-b3cf-89346b0e3ae9)
"Repair using"
![desc1](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/e11266f2-9a21-4916-ba4d-38dff482a719)

In other blocks, like "Disassembly ... might yield" and "You could use it to craft", it is displayed correctly.

**Comment by Uwuewsky** [comment]
I tried to do the same for ammunition. The text from the flag is not displayed in many places.
This is how the ammo name should be displayed (with `, reloaded [DEGUG]` (yes, degug)):

![e](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/2ac64158-af8c-4bc4-b969-74ce47a7ae9b)
---
And here's where it doesn't show up:
![c](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/f8929dce-93f6-4ee0-8cf6-483162d146a9)
Item suffix does not appear in container (magazine/gun):
![d](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/da1d698d-bd4c-4f2e-87ac-4b3f8e794368)
`f`iring menu:
![f](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/cb78db36-0faf-4cc5-b261-f864ef1ecb50)
`Variants` and `Ammunition`:
![variants](https://github.com/CleverRaven/Cataclysm-DDA/assets/99945024/7f72262a-759a-4ebf-855d-2bc27721e476)

**Comment by Brambor** [comment]
Yup, same issue. Let's finally confirm it so that it doesn't close again.

**Comment by Brambor** [comment]
- As far as I understand, with #78904, tackling this issue shouldn't be painful anymore.
